File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int l = 0, r = nums.size() - 1;
        int result = -1e9;
        while(l < r) {
            result = max(result, nums[l++] + nums[r--]);
        }
        return result;
    }
};

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.rs
impl Solution {
    pub fn min_pair_sum(mut nums: Vec<i32>) -> i32 {
        let mut res = 0;
        let n = nums.len();
        nums.sort();
        for i in 0..(n / 2) {
            res = res.max(nums[i] + nums[n - i - 1]);
        }
        res
    }
}